devito/types/sparse.py

```diff
@@ -132,7 +132,22 @@ def __distributor_setup__(self, **kwargs):
 
         return distributor
 
-    def __subfunc_setup__(self, key, suffix, dtype=None):
+    def __subfunc_setup__(self, suffix, keys, dtype=None, inkwargs=False, **kwargs):
+        key = None
+        for k in keys:
+            if k in kwargs:
+                if kwargs[k] is not None:
+                    key = kwargs[k]
+                    break
+                else:
+                    # In cases such as rebuild,
+                    # the subfunction may be passed explicitly as None
+                    return None
+        else:
+            if inkwargs:
+                # Only create the subfunction if provided.
```

Comment:
not sure I understand what the role of this argument is, perhaps expand the comment?